pkg/extension/sumologicextension/extension.go

```diff
@@ -294,13 +303,32 @@ func (se *SumologicExtension) registerCollector(ctx context.Context, collectorNa
 		zap.String(collectorCredentialIdField, resp.CollectorCredentialId),
 		zap.String(collectorCredentialKeyField, resp.CollectorCredentialKey),
 	)
-
 	return CollectorCredentials{
 		CollectorName: collectorName,
 		Credentials:   resp,
 	}, nil
 }
 
+// callRegisterWithBackoff calls registration using exponential backoff algorithm
+// this loosely base on backoff.Retry function
+func (se *SumologicExtension) registerCollectorWithBackoff(ctx context.Context, collectorName string) (CollectorCredentials, error) {
```

Comment:
Would it be possible to write a test for this? Shouldn't we have one?

Reply:
I added happy path test for extension startup

Reply:
That's a start 👍